src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -83,20 +93,52 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
             // Description
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
```

Comment:
Can group all the `if (displayAttribute != null)` logic instead of rechecking if displayAttribute is null every property.

Wish there was more we could do to reduce duplication but other routes are pretty ugly :cry: